specfile/specfile.py

```diff
@@ -0,0 +1,106 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import contextlib
+import types
+from pathlib import Path
+from typing import Iterator, List, Optional, Tuple, Type, Union
+
+from specfile.rpm import RPM, Macros
+from specfile.sections import Sections
+from specfile.tags import Tags
+
+
+class Specfile:
+    """
+    Class representing spec file.
+
+    Attributes:
+        path: Path to the spec file.
+        sourcedir: Path to sources and patches.
+        autosave: Whether to automatically save any changes made.
+        macros: List of macro definitions.
```

Comment:
Suggestion: `extra macro definitions` to stress that this is something else than the macros defined in the specfile.